content/docs/guides/continuous-delivery/troubleshooting-guide.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Troubleshooting Pulumi in CI
+meta_desc: This page walks-through the common failures encountered while running Pulumi in CI, as well as tips on how to fix them.
+menu:
+    userguides:
+        parent: cont_delivery
+        weight: 2
+---
+
+In order to understand the errors encountered during an automated pipeline execution, it is important to understand
+the steps involved in a typical CI configuration for Pulumi regardless of the CI service you are using.
+The type of failure you experience is likely related to one of these steps.
+
+## Overall Requirements
+
+In order to run a Pulumi command, the following are the requirements for any configuration involving Pulumi:
+
+* A Pulumi Access Token for the account you wish to use.
+Create one [here](https://app.pulumi.com/account/tokens) by logging in with the appropriate account.
+* A stack that you would like to update the automated pipeline.
+* Build tools (more on this below) based on the runtime of your Pulumi app.
+* Dependencies for your Pulumi app.
+* The right cloud provider credentials.
+
```

Comment:
Worth mentioning that Pulumi itself should be installed? Perhaps obvious, but could be a good opportunity to describe or link off to the proper ways of installing Pulumi in CI, mention that Pulumi comes pre-installed on GitHub Actions builders, etc.

Reply:
Yes, absolutely! Completely forgot that.

Reply:
I added an item here about the Pulumi CLI, as well as a section about it.